### PR TITLE
Changes for https://git.wieni.be/wieni/backend/-/issues/38

### PR DESCRIPTION
**Pull request by akasake**
## Description

- Restrict access to preview modal route [#12](https://github.com/wieni/wmmeta/issues/12)
- Not all required OG tags are included [#9](https://github.com/wieni/wmmeta/issues/9) 
- Clean up module [#6](https://github.com/wieni/wmmeta/issues/6)
- Unpublish date should be after publish date [#5](https://github.com/wieni/wmmeta/issues/5) 
- Lower the amount of logs by the scheduler [#15](https://github.com/wieni/wmmeta/issues/15)

### CODE REVIEW

**Comment by RobinHoutevelts** [comment]
LGTM 👍

File: src/Plugin/Validation/Constraint/TimeframeConstraint.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @Constraint(
+ *     id = "timeframe_constraint",
+ *     label = @Translation("Timeframe", context = "Validation"),
+ *     type = "entity:meta"
+ * )
+ */
+class TimeframeConstraint extends CompositeConstraintBase
+{
+    public $invalidTimeframe = 'The publish and unpublish dates do not form a valid timeframe';
```
Comment (RobinHoutevelts) [inline src/Plugin/Validation/Constraint/TimeframeConstraint.php:17]: Can't we reword this to `The unpublish date has to be after the publish date`.

"do not form a valid timeframe" might confuse editors

**Review by RobinHoutevelts — CHANGES_REQUESTED** [review]
After checking this out again, I do have a few minor request changes:

File: src/Plugin/Validation/Constraint/TimeframeConstraint.php (hunk 1 above)
Comment (akasake) [reply src/Plugin/Validation/Constraint/TimeframeConstraint.php]: Oke, this has been changed

**Comment by akasake** [comment]
@RobinHoutevelts I have applied you change requests. Could you have another look please?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 This changelog is incomplete. Pull requests with entries before 1.10.0
 are welcome.
 
+## [1.10.3] - 2023-02-xx
```
Comment (RobinHoutevelts) [inline CHANGELOG.md:10]:
```suggestion
## [1.10.4] - 2023-02-xx
```

File: wmmeta.install [hunk 3]
```diff
@@ -103,3 +103,19 @@ function wmmeta_update_8002(): void
...
 }
+
+/**
+ * Allow all roles to access wmmeta preview
+ * except anonymous and authenticated
+ */
+function wmmedia_update_8009(): void
+{
+    $storage = \Drupal::entityTypeManager()->getStorage('user_role');
+    $ids = $storage->getQuery()->condition('id', ['anonymous', 'authenticated'], 'NOT IN')->execute();
```
Comment (RobinHoutevelts) [inline wmmeta.install:114]: needs an `->accessCheck(false)`

**Review by RobinHoutevelts — CHANGES_REQUESTED** [review]
@akasake looking good!
Only a very minor change request.

After that you are good to merge! 👍

File: wmmeta.install (hunk 3 above)
Comment (akasake) [reply wmmeta.install]: hmmm I keep forgetting that 😮‍💨
